### PR TITLE
Keep Flutter.framework binaries writable so they can be code signed

### LINKED ISSUE #39507: Cannot codesign Flutter.framework, replacing existing signature, Permission denied
## Steps to Reproduce

```
cd dev/
SHARD=add2app_test dart bots/test.dart
```

## Logs
https://cirrus-ci.com/task/6430807339565056?command=test_all#L54
```
total size is 94542653  speedup is 1.00
Stripped /Users/m/Library/Developer/Xcode/DerivedData/ios_add2app-btprswwixqqclofvuogmdxvymovu/Build/Products/Debug-iphonesimulator/ios_add2app.app/Frameworks/Flutter.framework/Flutter of architectures: armv7 arm64
Code Signing /Users/m/Library/Developer/Xcode/DerivedData/ios_add2app-btprswwixqqclofvuogmdxvymovu/Build/Products/Debug-iphonesimulator/ios_add2app.app/Frameworks/Flutter.framework with Identity -
/usr/bin/codesign --force --sign -  --preserve-metadata=identifier,entitlements '/Users/m/Library/Developer/Xcode/DerivedData/ios_add2app-btprswwixqqclofvuogmdxvymovu/Build/Products/Debug-iphonesimulator/ios_add2app.app/Frameworks/Flutter.framework'
/Users/m/Library/Developer/Xcode/DerivedData/ios_add2app-btprswwixqqclofvuogmdxvymovu/Build/Products/Debug-iphonesimulator/ios_add2app.app/Frameworks/Flutter.framework: replacing existing signature
/Users/m/Library/Developer/Xcode/DerivedData/ios_add2app-btprswwixqqclofvuogmdxvymovu/Build/Products/Debug-iphonesimulator/ios_add2app.app/Frameworks/Flutter.framework: Permission denied
Command PhaseScriptExecution failed with a nonzero exit code


Test session results, code coverage, and logs:
	/Users/m/Library/Developer/Xcode/DerivedData/ios_add2app-btprswwixqqclofvuogmdxvymovu/Logs/Test/Test-ios_add2appTests-2019.08.29_11-37-54--0700.xcresult

Testing failed:
	Command PhaseScriptExecution failed with a nonzero exit code
	Testing cancelled because the build failed.

** TEST FAILED **


The following build commands failed:
	PhaseScriptExecution [CP]\ Embed\ Pods\ Frameworks /Users/magder/Library/Developer/Xcode/DerivedData/ios_add2app-btprswwixqqclofvuogmdxvymovu/Build/Intermediates.noindex/ios_add2app.build/Debug-iphonesimulator/ios_add2app.build/Script-7FADF19EC61F97E525982780.sh
(1 failure)
```

### PR DESCRIPTION
**Pull request by jmagman**
## Description

Flutter.framework files were made read-only to "Provide a strong hint to developers that editing Flutter framework headers isn't supported" (https://github.com/flutter/flutter/commit/cb2b89c389e06c2cceb1a3361d520eaccb4cec8c)  However this is making the Flutter binary readonly, which is being copied into the final embedded Frameworks directories, then can't be codesigned.

Only make headers, modulemaps, and plists readonly.

```shell
$ find Flutter/Flutter.framework -type f \( -name '*.h' -o -name '*.modulemap' -o -name '*.plist' \)   
Flutter/Flutter.framework/Headers/FlutterEngine.h
Flutter/Flutter.framework/Headers/FlutterChannels.h
Flutter/Flutter.framework/Headers/FlutterPlugin.h
Flutter/Flutter.framework/Headers/FlutterAppDelegate.h
Flutter/Flutter.framework/Headers/FlutterTexture.h
Flutter/Flutter.framework/Headers/FlutterPlatformViews.h
Flutter/Flutter.framework/Headers/FlutterHeadlessDartRunner.h
Flutter/Flutter.framework/Headers/FlutterCodecs.h
Flutter/Flutter.framework/Headers/Flutter.h
Flutter/Flutter.framework/Headers/FlutterViewController.h
Flutter/Flutter.framework/Headers/FlutterMacros.h
Flutter/Flutter.framework/Headers/FlutterDartProject.h
Flutter/Flutter.framework/Headers/FlutterPluginAppLifeCycleDelegate.h
Flutter/Flutter.framework/Headers/FlutterBinaryMessenger.h
Flutter/Flutter.framework/Headers/FlutterCallbackCache.h
Flutter/Flutter.framework/Modules/module.modulemap
Flutter/Flutter.framework/Info.plist
```

## Related Issues

Fixes https://github.com/flutter/flutter/issues/39507. 
Another attempt at https://github.com/flutter/flutter/pull/39539.

## Tests

add2app gets past codesigning for me, but it's still flaky so I didn't turn it back on.  See https://github.com/flutter/flutter/issues/40173.

## Checklist
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change
- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

### CODE REVIEW

**Review by christopherfujino — APPROVED** [review]
This _SEEMS_ foolproof...LGTM

File: packages/flutter_tools/bin/xcode_backend.sh [hunk 1]
```diff
@@ -142,11 +142,13 @@ BuildApp() {
     mkdir "${derived_dir}/engine"
     RunCommand cp -r -- "${flutter_podspec}" "${derived_dir}/engine"
     RunCommand cp -r -- "${flutter_framework}" "${derived_dir}/engine"
-    RunCommand find "${derived_dir}/engine/Flutter.framework" -type f -exec chmod a-w "{}" \;
+    # Make headers, plists, and modulemap files read-only to discourage editing.
+    RunCommand find "${derived_dir}/engine/Flutter.framework" -type f \( -name '*.h' -o -name '*.modulemap' -o -name '*.plist' \) -exec chmod a-w "{}" \;
```
Comment (jmagman) [inline packages/flutter_tools/bin/xcode_backend.sh]: To inline this, the last attempt at https://github.com/flutter/flutter/pull/39539/ was:
```
RunCommand find "${derived_dir}/engine/Flutter.framework" -type f -exec grep -Il '.' {} \; | xargs chmod a-w
```

I really don't know how this caused devicelab issues removing the repo, but this change is closer to what was there before (no `grep -I`), just adds checks the extensions.

**Comment by yrom** [comment]
hi @jmagman , I think the `stable` branch need this PR too, related issue #40146

**Comment by jmagman** [comment]
> hi @jmagman , I think the `stable` branch need this PR too, related issue #40146

Hi @yrom! You must be on the master channel to use Flutter modules since they are still an experimental feature. If you could `flutter update` and test this change, it would be greatly appreciated!

**Comment by OHeroJ** [comment]
> > hi @jmagman , I think the `stable` branch need this PR too, related issue #40146
> 
> Hi @yrom! You must be on the master channel to use Flutter modules since they are still an experimental feature. If you could `flutter update` and test this change, it would be greatly appreciated!

Means that the stable version under the existing app mixture is unable to use? My god！

**Comment by jmagman** [comment]
> > > hi @jmagman , I think the `stable` branch need this PR too, related issue #40146
> > 
> > 
> > Hi @yrom! You must be on the master channel to use Flutter modules since they are still an experimental feature. If you could `flutter update` and test this change, it would be greatly appreciated!
> 
> Means that the stable version under the existing app mixture is unable to use? My god！

You can see the caveats here: https://github.com/flutter/flutter/wiki/Add-Flutter-to-existing-apps#intro
> The "add-to-app" support is in preview, and is so far only available on the master channel.

> Since Flutter's "Add-to-App" functionality is in preview, the associated APIs and tooling are not stable and are subject to change.

**Comment by OHeroJ** [comment]
> > > > hi @jmagman , I think the `stable` branch need this PR too, related issue #40146
> > > 
> > > 
> > > Hi @yrom! You must be on the master channel to use Flutter modules since they are still an experimental feature. If you could `flutter update` and test this change, it would be greatly appreciated!
> > 
> > 
> > Means that the stable version under the existing app mixture is unable to use? My god！
> 
> You can see the caveats here: https://github.com/flutter/flutter/wiki/Add-Flutter-to-existing-apps#intro
> 
> > The "add-to-app" support is in preview, and is so far only available on the master channel.
> 
> > Since Flutter's "Add-to-App" functionality is in preview, the associated APIs and tooling are not stable and are subject to change.

but  run  `flutter build ios`, the same error!

**Comment by OHeroJ** [comment]
> > > > > hi @jmagman , I think the `stable` branch need this PR too, related issue #40146
> > > > 
> > > > 
> > > > Hi @yrom! You must be on the master channel to use Flutter modules since they are still an experimental feature. If you could `flutter update` and test this change, it would be greatly appreciated!
> > > 
> > > 
> > > Means that the stable version under the existing app mixture is unable to use? My god！
> > 
> > 
> > You can see the caveats here: https://github.com/flutter/flutter/wiki/Add-Flutter-to-existing-apps#intro
> > > The "add-to-app" support is in preview, and is so far only available on the master channel.
> > 
> > 
> > > Since Flutter's "Add-to-App" functionality is in preview, the associated APIs and tooling are not stable and are subject to change.
> 
> but run `flutter build ios`, the same error!

```sh
$ flutter build ios

Building com.mtd.iosFlutterModule for device (ios-release)...
Automatically signing iOS for device deployment using specified development team in Xcode project: BB535CKPZQ
Running Xcode build...

Xcode build done.                                           11.2s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **


Xcode's output:
↳
    /Users/oheroj/Desktop/TN/flutter_module/build/ios/Release-iphoneos/Runner.app/Frameworks/Flutter.framework: replacing existing
    signature
    /Users/oheroj/Desktop/TN/flutter_module/build/ios/Release-iphoneos/Runner.app/Frameworks/Flutter.framework: Permission denied
    Command CodeSign failed with a nonzero exit code
    note: Using new build systemnote: Planning buildnote: Constructing build description
```

**Comment by yrom** [comment]
@jmagman yep, our team were working with this 'experimental feature' in the stable branch and bring it to our online app on previous stable version (v1.7.8). 😂And we will apply this PR  to our inner fork version. thx.

**Comment by OHeroJ** [comment]
> @jmagman yep, our team were working with this 'experimental feature' in the stable branch and bring it to our online app on previous stable version (v1.7.8). 😂And we will apply this PR to our inner fork version. thx.

hello，If 1.9 stable continue to have a hotfix, the inner fork verion merge this hotfix branch?

**Pull request merged**